util/actions.js

```diff
@@ -0,0 +1,27 @@
+'use server';
+import { cookies } from 'next/headers';
+import { getFruitByName } from '../database/fruits';
+import { getCookieByName } from './cookies';
+
+export async function setFruitNote(formData) {
+  const singleFruit = getFruitByName(formData.get('fruit-name'));
+
+  if (!singleFruit) return;
+
+  const userFruitNotes = getCookieByName('fruitNotes');
+
+  const oldNotes = Array.isArray(userFruitNotes)
+    ? userFruitNotes.filter((fruitNote) => fruitNote.id !== singleFruit.id)
+    : [];
+
+  await cookies().set(
+    'fruitNotes',
+    JSON.stringify([
+      ...oldNotes,
+      {
+        id: singleFruit.id,
+        note: formData.get('fruit-note'),
+      },
+    ]),
```

Comment:
this reorders the list - let's avoid this reordering so that students don't have that in their project (they will definitely copy this)

it does mean slightly more complex code, but it's worth it

Reply:
https://github.com/upleveled/next-js-example-winter-2023-vienna-austria/blob/update-fruit-example-cookies/app/fruits/%5BfruitId%5D/actions.js